context-monitoring/src/main/java/de/atb/context/monitoring/monitors/file/FileSystemMonitor.java

```diff
@@ -181,18 +172,21 @@ protected final void startWatcher() {
 
             @Override
             public void run() {
-                WatchService watchService = FileSystems.getDefault().newWatchService();
+                WatchService watchService = null;
```

Comment:
There seems to be a lot of duplicated code in the three file monitor classes, which could be extracted into some form of shared code.

Maybe this could be a good moment to cleanup the code.

Reply:
done.
Extracted common code into AbstractFileSystem monitor